webapi/src/main/scala/org/knora/webapi/util/ConstructResponseUtilV2.scala

```diff
@@ -1215,97 +1242,56 @@ object ConstructResponseUtilV2 {
-      * @return a collection of [[ReadResourceV2]] representing the search results.
-      */
-    def createSearchResponse(searchResults: RdfResources,
-                             orderByResourceIri: Seq[IRI],
-                             mappings: Map[IRI, MappingAndXSLTransformation] = Map.empty[IRI, MappingAndXSLTransformation],
-                             queryStandoff: Boolean,
-                             forbiddenResource: Option[ReadResourceV2],
-                             responderManager: ActorRef,
-                             targetSchema: ApiV2Schema,
-                             settings: SettingsImpl,
-                             requestingUser: UserADM)(implicit stringFormatter: StringFormatter, timeout: Timeout, executionContext: ExecutionContext): Future[Vector[ReadResourceV2]] = {
-
-        if (orderByResourceIri.toSet != searchResults.keySet && forbiddenResource.isEmpty) throw AssertionException(s"Not all resources are visible, but forbiddenResource is None")
+     * Creates an API response.
+     *
+     * @param mainResourcesAndValueRdfData the query results.
+     * @param orderByResourceIri           the order in which the resources should be returned.
+     * @param mappings                     the mappings to convert standoff to XML, if any.
+     * @param queryStandoff                if `true`, make separate queries to get the standoff for text values.
+     * @param versionDate                  if defined, represents the requested time in the the resources' version history.
+     * @param responderManager             the Knora responder manager.
+     * @param targetSchema                 the schema of response.
+     * @param settings                     the application's settings.
+     * @param requestingUser               the user making the request.
+     * @return a collection of [[ReadResourceV2]] representing the search results.
+     */
+    def createApiResponse(mainResourcesAndValueRdfData: MainResourcesAndValueRdfData,
+                          orderByResourceIri: Seq[IRI],
+                          mappings: Map[IRI, MappingAndXSLTransformation] = Map.empty[IRI, MappingAndXSLTransformation],
+                          queryStandoff: Boolean,
+                          calculateMayHaveMoreResults: Boolean,
+                          versionDate: Option[Instant],
+                          responderManager: ActorRef,
+                          targetSchema: ApiV2Schema,
+                          settings: SettingsImpl,
+                          requestingUser: UserADM)(implicit stringFormatter: StringFormatter, timeout: Timeout, executionContext: ExecutionContext): Future[ReadResourcesSequenceV2] = {
+
+        val visibleResourceIris: Seq[IRI] = orderByResourceIri.filter(resourceIri => mainResourcesAndValueRdfData.resources.keySet.contains(resourceIri))
 
         // iterate over orderByResourceIris and construct the response in the correct order
```

Comment:
update the comment 
iterate over `orderByResourceIris` should be iterate over `visibleResourceIris`

Reply:
Done in 6e96460, thanks.